### PR TITLE
docs(navigation): update vanilla architecture and navigation article

### PR DESCRIPTION
**Pull request by ADjenkov**
Fix and Update of code snippets (to be more copy pastable).

### CODE REVIEW

File: docs/core-concepts/navigation.md [hunk 1]
```diff
@@ -157,25 +158,32 @@ A more dynamic way of navigating can be done by providing a function that return
...
+const factoryFunc = (): Page => {
+    const label = new Label();
     label.text = "Hello, world!";
-    var page = new pagesModule.Page();
+    const page = new Page();
     page.content = label;
     return page;
 };
-topmost.navigate(factoryFunc);
+topmost().navigate(factoryFunc);
```
Comment (tjvantoll) [inline docs/core-concepts/navigation.md:186]: Wow, good catch!

Comment (tjvantoll) [reply docs/core-concepts/navigation.md:186]: Oh I see. This worked before, but the previous code was, uh, weird 😛

Comment (ADjenkov) [reply docs/core-concepts/navigation.md:186]: Exactly! I personally think that this way it's more obvious how it should be used

**Comment by tjvantoll** [comment]
Merged with 2a9adf5ce33830dfa886b69db6c624c46af14ecb. Thanks so much @ADjenkov! This really cleans this article up.

**Pull request closed without merging**